packages/core/src/dispatcher.ts

```diff
@@ -213,12 +244,18 @@ export default class Dispatcher extends EventEmitter {
    * @private
    */
   async _handleQueryMessage(message: any): Promise<void> {
-    const { id } = message
-    if (!this._documents[id]) {
+    // Runtime check that the message adheres to the expected format for QUERY messages
+    QueryMessage.check(message)
+
+    const { doc } = message
+    if (!this._documents[doc]) {
       return
     }
 
-    this._documents[id].emit('tipreq')
+    // TODO: Should we validate that the 'id' field is the correct hash of the rest of the message?
```

Comment:
I'm wondering if it's worth it to validate that the 'id' property is what we expect it to be.  I'm leaning towards no, at least for now, as we don't actually *rely* on the id format for protocol correctness, it's just an optimization to allow deduping query messages.  Curious your thoughts on this though @oed @simonovic86